viewing_party/party.py

```diff
@@ -0,0 +1,215 @@
+#
+#Robot invasion wave 1
+#
+from typing import Counter
+from statistics import mode
+
+def create_movie(title,genre,rating):
+    if bool(title) and bool(genre) and  bool(rating):
+        movie= {"title": title, "genre":genre, "rating": rating}
+        return movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie) 
+    return user_data
+  
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data,title):
+
+    for i,dct in enumerate(user_data["watchlist"]):
+        if title == dct["title"]:
+                watched_movie_dct = user_data["watchlist"].pop(i)
+                user_data["watched"].append(watched_movie_dct)
+                return user_data
+    return user_data
+    
+#
+#Robot invasion Wave 2
+#
+def get_watched_avg_rating(user_data):
+    sum=0
+    if user_data["watched"]==[]:
+        return 0
+    
+    for dct in user_data["watched"]:
+        sum += dct["rating"]
+    return sum/len(user_data["watched"])
+
+def get_most_watched_genre(user_data):
+    most_watched_lst=[]
+
+    if user_data["watched"]==[]:
+        return None
+    for mvie in user_data["watched"]:
+        most_watched_lst.append(mvie["genre"])
+    genre_cnt = Counter(most_watched_lst)
```

Comment:
Nice work researching and implementing these tools for collections.

A common way to solve this problem without is to use a frequency map: https://www.geeksforgeeks.org/counting-the-frequencies-in-a-list-using-dictionary-in-python/. I encourage you to review this method and consider using in future projects.